sdk/java/pulumi/src/test/java/com/pulumi/deployment/DeploymentRunnerTest.java

```diff
@@ -31,43 +31,34 @@ public void cleanup() {
         var mock = DeploymentTests.DeploymentMockBuilder.builder()
-            .setMocks(new MocksTest.MyMocks())
-            .setSpyGlobalInstance();
+                .setMocks(new MocksTest.MyMocks())
+                .setSpyGlobalInstance();
 
         mock.standardLogger.setLevel(Level.OFF);
-        var result = mock.tryTestAsync(TerminatesEarlyOnExceptionStack::init).join();
-
-        assertThat(mock.runner.getSwallowedExceptions()).hasSize(2);
+        var result = mock.runTestAsync(ctx -> {
+            Output.of(CompletableFuture.failedFuture(new RunException("Deliberate test error")));
+            ctx.export("slowOutput", Output.of(
+                    new CompletableFuture<Integer>()
+                            .completeOnTimeout(1, 60, TimeUnit.SECONDS)
+            ));
+        }).join();
+
+        assertThat(result.errors).isNotNull();
+        assertThat(result.errors).isNotEmpty();
+        assertThat(result.errors).hasSize(1);
+        assertThat(result.errors).haveAtLeastOne(containsString("Deliberate test error"));
+
+        assertThat(result.exceptions).hasSize(2);
+        assertThat(result.exceptions.get(0)).isExactlyInstanceOf(CompletionException.class);
+        assertThat(result.exceptions.get(1)).isExactlyInstanceOf(RunException.class);
+        assertThat(result.exceptions.get(1)).hasMessageContaining("Deliberate test error");
 
-        assertThat(result.exceptions).isNotNull();
-        assertThat(result.exceptions).isNotEmpty();
-        assertThat(result.exceptions).hasSize(1);
-        assertThat(result.exceptions.get(0)).isExactlyInstanceOf(RunException.class);
-        assertThat(result.exceptions.get(0)).hasMessageContaining("Deliberate test error");
         assertThat(result.resources).isNotNull();
         assertThat(result.resources).isNotEmpty();
         assertThat(result.resources).hasSize(1);
-        var stack = TerminatesEarlyOnExceptionStack.of(result);
-        assertThat(Internal.of(stack.slowOutput).getDataAsync()).isNotCompleted();
-        assertThat(Internal.of(stack.slowOutput).getValueNullable()).isNotCompleted();
-    }
-
-    public static class TerminatesEarlyOnExceptionStack {
-        public final Output<Integer> slowOutput;
-
-        public TerminatesEarlyOnExceptionStack(Output<Integer> slowOutput) {
-            this.slowOutput = slowOutput;
-        }
-
-        public static void init(Context ctx) {
-            Output.of(CompletableFuture.failedFuture(new RunException("Deliberate test error")));
-            ctx.export("slowOutput",Output.of(new CompletableFuture<Integer>()
-                    .completeOnTimeout(1, 60, TimeUnit.SECONDS)));
-        }
-
-        public static TerminatesEarlyOnExceptionStack of(DeploymentTests.DeploymentMock.TestAsyncResult result) {
-            return new TerminatesEarlyOnExceptionStack(result.getStackOutput("slowOutput"));
-        }
+        assertThat(Internal.of(result.stackOutput("slowOutput")).getDataAsync()).isNotCompleted();
```

Comment:
This thinks that `result.stackOutput` is `Output<Object>` or `Output<?>` right? But that seems OK since the test does not care? This seems reasonable in this test.

Reply:
yes: `public Output<Object> stackOutput(String name)`